### PR TITLE
distsql: improve package distsql test code coverage to 85%

### PR DESCRIPTION
**Pull request by XuHuaiyu**
### What problem does this PR solve? 
Improve package distsql test coverage to 85%.

### What is changed and how it works?
before:
`coverage: 68.4% of statements`

after:
`coverage: 85.1% of statements`

### CODE REVIEW

File: distsql/c.out [hunk 1]
```diff
@@ -0,0 +1,220 @@
+mode: set
```
Comment (jackysp) [inline distsql/c.out]: Is this file necessary?

File: distsql/select_result_test.go [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package distsql
+
+import (
+	"fmt"
```
Comment (lamxTyler) [inline distsql/select_result_test.go]: Leave a blank line?

File: distsql/stream.go [hunk 3]
```diff
@@ -123,10 +123,9 @@ func (r *streamResult) flushToChunk(chk *chunk.Chunk) (err error) {
 			}
 		}
 	}
+	r.curr.RowsData = remainRowsData
```
Comment (zz-jason) [inline distsql/stream.go]: why changing this?

Comment (XuHuaiyu) [reply distsql/stream.go]: For improving the test coverage.
Before this, the else branch will not be touched.

File: distsql/select_result_test.go [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+	sr.updateCopRuntimeStats("callee")
+
+	sr.copPlanIDs = []fmt.Stringer{copPlan{}}
+	c.Assert(ctx.GetSessionVars().StmtCtx.RuntimeStatsColl, NotNil)
+	c.Assert(len(sr.selectResp.GetExecutionSummaries()), Equals, len(sr.copPlanIDs))
+	sr.updateCopRuntimeStats("callee")
+	for _, execSum := range sr.selectResp.GetExecutionSummaries() {
+		c.Assert(*execSum.NumIterations, Equals, uint64(1))
+		c.Assert(*execSum.NumProducedRows, Equals, uint64(1))
+		c.Assert(*execSum.TimeProcessedNs, Equals, uint64(1))
```
Comment (eurekaka) [inline distsql/select_result_test.go]: Should we check the values in `ctx.GetSessionVars().StmtCtx.RuntimeStatsColl` instead of `sr.selectResp.GetExecutionSummaries()`?

File: distsql/select_result_test.go [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+
+	ctx.GetSessionVars().StmtCtx.RuntimeStatsColl = execdetails.NewRuntimeStatsColl()
+	t := uint64(1)
+	sr.selectResp = &tipb.SelectResponse{
+		ExecutionSummaries: []*tipb.ExecutorExecutionSummary{
+			{TimeProcessedNs: &t, NumProducedRows: &t, NumIterations: &t},
+		},
+	}
+	c.Assert(len(sr.selectResp.GetExecutionSummaries()) != len(sr.copPlanIDs), IsTrue)
+	sr.updateCopRuntimeStats("callee")
```
Comment (eurekaka) [inline distsql/select_result_test.go]: Check `ctx.GetSessionVars().StmtCtx.RuntimeStatsColl` is not updated?
